ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1736,6 +1768,10 @@ public boolean validateMaterializedViewsFromRegistry(List<Table> cachedMateriali
       // Final result
       boolean result = true;
       for (Table cachedMaterializedViewTable : cachedMaterializedViewTables) {
+        if (!checkPrivillegeForMV(cachedMaterializedViewTable)) {
```

Comment:
`validateMaterializedViewsFromRegistry` is only called if the MV is coming from the registry. However, we need the authorization check in all cases, e.g., dummy registry.
You can simply call the new method from Calcite planner.